frontend/src/scenes/feature-flags/FeatureFlagEvaluationTags.tsx

```diff
@@ -0,0 +1,178 @@
+    tagsAvailable,
+    staticOnly = false,
+    className,
+}: FeatureFlagEvaluationTagsProps): JSX.Element {
+    const logic = featureFlagEvaluationTagsLogic({ tags, evaluationTags })
+    const { editingTags, showEvaluationOptions, selectedTags, selectedEvaluationTags } = useValues(logic)
+    const {
+        setEditingTags,
+        setShowEvaluationOptions,
+        setSelectedTags,
+        setSelectedEvaluationTags,
+        updatePropsAndReset,
+    } = useActions(logic)
+
+    const { saveFeatureFlag } = useActions(featureFlagLogic)
+    const { featureFlagLoading } = useValues(featureFlagLogic)
+
+    // Keep selections fresh when props change while not editing
+    // Check if props have changed and update logic accordingly
+    const currentTagsString = selectedTags.join(',')
+    const currentEvalTagsString = selectedEvaluationTags.join(',')
+    const propsTagsString = tags.join(',')
+    const propsEvalTagsString = evaluationTags.join(',')
+
+    if (!editingTags && (currentTagsString !== propsTagsString || currentEvalTagsString !== propsEvalTagsString)) {
+        updatePropsAndReset(tags, evaluationTags)
+    }
+
+    const handleSave = (): void => {
+        if (onChange) {
+            onChange(selectedTags, selectedEvaluationTags)
+        } else {
+            saveFeatureFlag({
+                tags: selectedTags,
+                evaluation_tags: selectedEvaluationTags,
+            })
+        }
+        setEditingTags(false)
+        setShowEvaluationOptions(false)
+    }
+
+    const toggleEvaluationTag = (tag: string): void => {
+        if (selectedEvaluationTags.includes(tag)) {
+            setSelectedEvaluationTags(selectedEvaluationTags.filter((t: string) => t !== tag))
+        } else {
+            setSelectedEvaluationTags([...selectedEvaluationTags, tag])
+        }
+    }
+
+    if (editingTags) {
+        return (
+            <div className={clsx(className, 'flex flex-col gap-2')}>
+                <div className="flex items-center gap-2">
+                    <LemonInputSelect
+                        mode="multiple"
+                        allowCustomValues
+                        value={selectedTags}
+                        options={tagsAvailable?.map((t: string) => ({ key: t, label: t }))}
+                        onChange={setSelectedTags}
+                        loading={featureFlagLoading}
+                        data-attr="feature-flag-tags-input"
+                        placeholder='Add tags like "production", "app", "docs-page"'
+                        autoFocus
+                    />
+                    {selectedTags.length > 0 && (
+                        <LemonButton
+                            size="small"
+                            onClick={() => setShowEvaluationOptions(!showEvaluationOptions)}
+                            icon={<IconBolt />}
+                            tooltip="Configure evaluation environments"
+                        />
+                    )}
```

Comment:
this will have a tooltip explaining what it does with a link to the docs once i write them.